chirp/drivers/radtel_rt493.py

```diff
@@ -0,0 +1,1021 @@
+
+    struct {
+        u8 scan_add1[16];       // "Scan Add" settings for channels 1-128
+        u8 scan_add2[16];       // "Scan Add" settings for channels 129-199
+    } scan_add_settings;
+
+    struct {
+        u8 learn_code1[16];     // "Learn Code" settings for channels 1-128
+        u8 learn_code2[16];     // "Learn Code" settings for channels 129-199
+    } learn_code_settings;
+
+    struct {
+        u8 power_level1[16];    // Power level settings (1-128)
+        u8 power_level2[16];    // Power level settings (129-199)
+    } power_settings;
+
+    u8 unknown1[16];            // Unknown data
+    u8 unknown2[16];            // Unknown data
+    """
+
+    # Return information about this radio's features, including
+    # how many memories it has, what bands it supports, etc
+    def get_features(self):
+        rf = chirp_common.RadioFeatures()
+        rf.has_settings = True
+        rf.has_bank = False
+        rf.has_rx_dtcs = True
+        rf.has_dtcs = True
+        rf.has_ctone = True
+        rf.has_dtcs_polarity = True
+        rf.has_name = False
+        rf.can_odd_split = True
+        rf.has_cross = True
+        rf.has_tuning_step = False
+        rf.has_variable_power = False
+        rf.has_offset = True
+        rf.has_mode = False
+
+        rf.valid_modes = ["NFM", "WFM"]
+        rf.valid_duplexes = ["", "-", "+", "split"]
+        rf.valid_tmodes = ["", "Tone", "TSQL", "DTCS", "Cross"]
+        rf.valid_skips = ["", "S"]
+        rf.valid_power_levels = self.POWER_LEVELS
+        rf.valid_cross_modes = [
+            "Tone->Tone",
+            "DTCS->",
+            "->DTCS",
+            "Tone->DTCS",
+            "DTCS->Tone",
+            "->Tone",
+            "DTCS->DTCS",
+            # "Tone->"
+        ]
+        rf.valid_dtcs_codes = self.DTCS_CODES
+        rf.memory_bounds = (1, 199)
+        # TODO: maybe limit to EU version (430000000, 440000000)
+        rf.valid_bands = [(400000000, 470000000)]
+        rf.valid_characters = chirp_common.CHARSET_ALPHANUMERIC + "-"
+        return rf
+
+    def _decode_qt_dqt(self, qt_bytes):
+        assert len(qt_bytes) == 2
+
+        value = int.from_bytes(qt_bytes, "little")
+        byte1 = qt_bytes[0]
+        byte2 = qt_bytes[1]
+
+        if value == 0xFFFF or value == 0x0000:
+            return None, None, None
+        elif byte2 <= 9:
+            if value / 10 in chirp_common.TONES:
+                return "Tone", value / 10, None
+            else:
+                raise ValueError("Invalid tone value: %i" % value)
+
+        # DTCS Normal Polarity (DxxxN) and DTCS Inverted Polarity (DxxxI)
+        dtcs_code_idx = _decode_dtcs_byte(byte1, byte2)
+        assert dtcs_code_idx != -1
+        dtcs_polarity = "N" if byte2 == 0x28 or byte2 == 0x29 else "R"
+        if 0 <= dtcs_code_idx <= len(self.DTCS_CODES) - 1:
+            return "DTCS", self.DTCS_CODES[dtcs_code_idx], dtcs_polarity
+        else:
+            return None, None, None
```

Comment:
This looks like you're decoding each byte independently and doing your own math. I think you could use the `union` functionality in bitwise to avoid a lot of this.